samples/core/resource_spec/runtime_resource_request_gpu.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2021 The Kubeflow Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import kfp
+from kfp import dsl, components
+from typing import NamedTuple
+
+
+def training_job():
+    import torch
+    use_cuda = torch.cuda.is_available()
+    print(f"The gpus status is: {use_cuda}")
+    if not use_cuda:
+        raise ValueError("GPU not available")
+
+
+training_comp = components.create_component_from_func(
+        training_job,
+        base_image='pytorch/pytorch:1.7.1-cuda11.0-cudnn8-runtime',
+        packages_to_install=[]
+    )
+
+@components.create_component_from_func
+def generate_resource_request() -> NamedTuple('output', [('gpu_vendor', str), ('nbr_gpus', str), ('constrain_type', str), ('constrain_value', str)]):
+    '''Returns the gpu resource settings'''
```

Comment:
nit: Use double quotes for docstring, and end the description with a period.

https://google.github.io/styleguide/pyguide.html#381-docstrings